craft_parts/sources/errors.py

```diff
@@ -58,6 +58,27 @@ def __init__(self, *, source_type: str, option: str):
         super().__init__(brief=brief, resolution=resolution)
 
 
+# TODO: Merge this with InvalidSourceOption above
```

Comment:
do you intend to do this on this PR?

Reply:
No, because it'd include touching a bunch of things not related to this. I've got another branch open for that though

Reply:
Well I was going to submit that PR tomorrow, but I came up with this after dinner as what might be a better way to do it generically. LMK what you think: https://github.com/canonical/craft-parts/pull/394/commits/015b8592d4c44f815560e45d5bfddede9f0e5ecc